docs/graph-homomorphism-as-natural-transformation.md

```diff
@@ -19,6 +23,12 @@ An example of this in haskell is Maybe.
  * The functor G(X) is Nothing.
  * The morphism between all F(X) and G(X) is Nothing (see [this issue though](https://github.com/matthewfranglen/category-set-graph/issues/1))
 
+[Paul: Let me translate this example into my language to see if I understand it properly.  The categories C and D are both *Set*, the functor F: *Set* -> *Set* is the identity (sends each object to itself and each morphism to itself), and the functor G: *Set* -> *Set* sends each set to the empty set and each function between sets to the empty function from the empty set to itself.  You propose to construct a natural transformation F -> G.
```

Comment:
So this requires dividing the target set of Maybe T into the target set Just T and the target set Nothing.

The target set Nothing is not the empty set, it is the singleton set. The only object in that set is Nothing. There is an extension in haskell which allows you to define the Void type, and that has no objects. The fact that I am able to construct an object that is in the target set Nothing means that it is not empty.

I think one of the confusions here is that the _value_ is called Nothing and the way you construct it is to call the completely separate and different constructor which is called Nothing. That constructor takes no arguments though so it's super confusing:

```
ghci ➜ :i Nothing
data Maybe a = Nothing | ...    -- Defined in ‘GHC.Base’
ghci ➜ :t Nothing
Nothing :: Maybe a
```